tutorials/environmental_data_and_visualization.md

```diff
@@ -0,0 +1,173 @@
+and water temperature one needs to have some way of defining values such that
+the values vary through out the environment. Gazebo's Environment Data
+infrastructure enables such visualization and sensing capabilities.
+
+# Loading Environment Data
+
+Currently gazebo is able to load data from CSV files and display them. It is
+assumed the points are sampled at a fixed grid interval. For gaps in between the
+sample points, Gazebo will interpolate the data.
+
+## Data Format
+
+As mentioned earlier, the data should be a CSV file with grid based spacing.
+The first line of the CSV file should be the heading for each column.
+There should be a column for time (in seconds), the x coordinate, y coordinate
+and z coordinate (in meters). You may have multiple columns of data like so:
+```
+timestamp,humidity,x,y,z
+0,0,-1,-1,-1
+0,0,-1,-1, 1
+0,0,-1, 1,-1
+0,0,-1, 1, 1
+0,0, 1,-1,-1
+0,0, 1,-1, 1
+0,0, 1, 1,-1
+0,0, 1, 1, 1
+1,1,-1,-1,-1
+1,1,-1,-1, 1
+1,1,-1, 1,-1
+1,1,-1, 1, 1
+1,1, 1,-1,-1
+1,1, 1,-1, 1
+1,1, 1, 1,-1
+1,1, 1, 1, 1
+
+```
+For the x and y coordinates you may also use latitude and longitude in degrees.
+Heading names don't matter. Order also doesn't matter.
+```
+timestamp,humidity,latitude,longitude,altitude
+1658923062,91,36.80029505,-121.788972517,0.8
+1658923062,88,36.80129505,-121.788972517,0.8
+1658923062,89,36.80029505,-121.789972517,0.8
+1658923062,92,36.80129505,-121.789972517,0.8
+1658923063,90,36.80029505,-121.788972517,0.8
+1658923063,85,36.80129505,-121.788972517,0.8
+1658923063,87,36.80029505,-121.789972517,0.8
+1658923063,94,36.80129505,-121.789972517,0.8
+```
+> :warning: Note that only one Environment Data file per world is allowed. Also,
+Gazebo can only load data that fits into memory.
+
+## Pre-Loading
+If you are writing SDFormat files by hand or via a script you may want to
+pre-load the data. You may use the `EnvironmentPreload` plugin for this. For
+instance we could load the first file like so (by adding the environment to the
+):
+```xml
+    <plugin
+      filename="gz-sim-environment-preload-system"
+      name="gz::sim::systems::EnvironmentPreload">
+      <data>environmental_data.csv</data>
+      <dimensions>
+        <time>timestamp</time>
+        <space>
+          <x>x</x>
+          <y>y</y>
+          <z>z</z>
+        </space>
+      </dimensions>
+    </plugin>
+```
+If you would like to use Latitude and Longitude you may set the `reference` to
+`spherical` and the `units` to `degrees`.
+```xml
+    <plugin
+      filename="gz-sim-environment-preload-system"
+      name="gz::sim::systems::EnvironmentPreload">
+      <data>environmental_data.csv</data>
+      <dimensions>
+        <time>timestamp</time>
+        <space reference="spherical" units="degrees">
+          <x>latitude</x>
+          <y>longitude</y>
+          <z>altitude</z>
+        </space>
+      </dimensions>
+    </plugin>
+```
+
+## Via GUI
+To load the data, click on the 3 dots and search for the "Environment Loader":
```

Comment:
For completeness, should instruct the user how to bring up the GUI we are starting from, e.g. I'm assuming it is
```
gz sim empty.sdf
```

Reply:
Ah, with the new changes we need to preload 
```
    <plugin
      filename="gz-sim-environmental-sensor-system"
      name="gz::sim::systems::EnvironmentalSystem">
    </plugin>
```
Otherwise nothing will work. Perhaps I should add an empty_environment_system.sdf?